cmd/init-osm-controller/init-osm-controller.go

```diff
@@ -112,9 +92,24 @@ func main() {
 		return
 	}
 
-	meshConfig := createDefaultMeshConfig()
+	presetMeshConfig, err := configClient.ConfigV1alpha1().MeshConfigs(osmNamespace).Get(context.TODO(), presetMeshConfigName, metav1.GetOptions{})
+	_, meshConfigExists := configClient.ConfigV1alpha1().MeshConfigs(osmNamespace).Get(context.TODO(), meshConfigName, metav1.GetOptions{})
+
+	// If the presetMeshConfig could not be loaded and a default meshConfig doesn't exist, return the error
+	if !apierrors.IsAlreadyExists(meshConfigExists) && err != nil {
```

Comment:
This check seems off to me. Why is there a call to `IsAlreadyExists()` with an argument corresponding to an error val returned from a Get()? Shouldn't an `already exists` error only be returned fo Create()?

Reply:
that's right I updated the error to `IsNotFound()` and extended out the e2e